openstudiocore/src/utilities/idf/IdfObject.cpp

```diff
@@ -1854,7 +1856,17 @@ void IdfObject_Impl::populateValidityReport(ValidityReport& report, bool checkNa
         else {
           return false;
         }
-      }
+      } else{
+        if (std::isnan(*value)) {
```

Comment:
@jmarrec this catches the "Infinity" or "NaN" in the field after it has been set but during validity checking, this will prevent issues where you bypass setDouble using the pushExtensibleGroup code

Reply:
The only issue is this code only runs when using it on WorkspaceObjects or ModelObjects, won't work on IdfObjects (but would work if you tried to add an IdfObject to a Workspace)